iceoryx_examples/icedelivery/iox_publisher_typed_modern.cpp

```diff
@@ -49,36 +49,44 @@ int main(int argc, char* argv[])
     {
         ++ct;
 
-        // Retrieve a typed sample from shared memory.
-        // Sample can be held until ready to publish.
+        // API Usage #1
+        //  * Retrieve a typed sample from shared memory.
+        //  * Sample can be held until ready to publish.
+        //
         auto result = typedPublisher.loan();
         if (!result.has_error())
         {
             auto& sample = result.get_value();
-            sample->x = ct * 1.1;
-            sample->y = ct * 1.1;
-            sample->z = ct * 1.1;
+            sample->x = ct;
+            sample->y = ct;
+            sample->z = ct;
             sample.publish();
         }
```

Comment:
Would an else branch make sense, e.g. printing an error?
Just asking because I think it would make sense with the lambda to show how to access the error, e.g.
```
typedPublisher.load()
    .and_then(...)
    .or_else(...) { // do something with the eroor}
```